cql3/restrictions/statement_restrictions.cc

```diff
@@ -615,19 +615,16 @@ std::vector<const column_definition*> statement_restrictions::get_column_defs_fo
                 }
             }
         }
-        auto single_ck_restrs = dynamic_pointer_cast<single_column_clustering_key_restrictions>(_clustering_columns_restrictions);
         const bool pk_has_unrestricted_components = has_partition_key_unrestricted_components();
         if (pk_has_unrestricted_components || clustering_key_restrictions_need_filtering()) {
             column_id first_filtering_id = pk_has_unrestricted_components ? 0 : _schema->clustering_key_columns().begin()->id +
                     num_clustering_prefix_columns_that_need_not_be_filtered();
             for (auto&& cdef : expr::get_sorted_column_defs(_new_clustering_columns_restrictions)) {
                 const expr::expression* single_col_restr = nullptr;
-                if (single_ck_restrs) {
-                    auto it = single_ck_restrs->restrictions().find(cdef);
-                    if (it != single_ck_restrs->restrictions().end()) {
-                        if (is_single_column_restriction(it->second->expression)) {
-                            single_col_restr = &it->second->expression;
-                        }
+                auto it = _single_column_clustering_key_restrictions.find(cdef);
+                if (it != _single_column_clustering_key_restrictions.end()) {
+                    if (is_single_column_restriction(it->second)) {
```

Comment:
Wait, you just looked this entry up from "single column clustering key restrictions map", how can it not be single column? Sounds like this condition is always true and can be removed